src/commands/push.js

```diff
@@ -103,12 +93,12 @@ function push(repo, options) {
   }
 
   return getRemoteAndBranchPromise
-    .then(() => core.push(`${remote} ${branch}`, { cwd: repoPath }, callback))
-    .then(({ stdout }) => {
-      response.raw = stdout;
-      response.push = generatePushStats(stdout);
-      return response;
-    }, errorCatchHandler(response));
+    .then(() => core.push(`${remote} ${branch}`, { cwd: repo.path() }, callback))
+    .then(({ stdout }) => ({
+      ...generateResponse(),
+      push: generatePushStats(stdout),
```

Comment:
I know you technically did not write this code, but it would be good to follow the same pattern as the other `spawn` based commands, where we look for `response.push.push_error` or whatever and if we find it we can set success to `false` and set the error message on `stderr`. Could you do that? See `pull.js` if you want an example, but I feel you know what I am talking about.